product/02-innersource-and-agile.md

```diff
@@ -0,0 +1,82 @@
+
+
+## Autonomous teams in an interconnected world
+
+* organisations try to deliver value to customers quickly
+* to do that lengthy roadmap synchronisation settings are to be avoided
+* often those discussions stem from team dependencies: Releases waiting for the testers, deployment waiting for time in the ops team.
+As a result, cross functional teams where testing happens close to development and those who build software own running the software have proven to be successful.
+* but there is another type of inter-team dependency:
+* however we live in an interconnected world: Software written today relies on many dependencies
+* Each source code dependency, each call to some service, each external dataset used boils down to a dependency between two teams.
+* What if you need changes made to such a dependency?
+   * Traditional options:
+      * submit a feature request and wait it out
+      * build a workaround
+      * if nothing else helps, escalate your issue
+   * added benefit of InnerSource:
+      * help with the solution
+
+* But team members working in each others repositories sounds like chaos?
+   * InnerSource comes with a set of roles and processes, contributions happen in a structured way
+   * each InnerSource project has a clear set of trusted committers with clear accountabilities that go beyond simply reviewing code.
+   * support for more autonomy by being able to fix upstream artifacts without sacrifycing quality due to a clear review process
+
+* Todays world moves towards a remote first culture - InnerSource has a focus on written communication and lends itself easily to remote first.
+
+
+## Does InnerSource replace Agile?
+
+* short answer: No, not at all.
+* Instead they complement each other:
+
+
+- How InnerSource and Agile complement each other
+   - Well factored and well tested code leads to faster on-ramp for agile teams, but also helps InnerSource allowing for faster on-ramp for contributors
+   - A focus on working together and not assigning individual tasks allows for easier on-ramp of contributors.
+   - Existing local pairing can make cross team pairing and sharing of work in progress easier.
+   - Existing test driven development means that confidence in contributions is higher.
+   - XP teams have a "leave the code in better shape than you found it" attitude - mitigates the risk of decline in quality or co-hesion of code when faced with multiple contributions from different sources.
+
+## Common misunderstandings when coming from agile teams
+
+* Impact on performance reviews: InnerSource moves some of the energy you spend developing outside of your own team.
+That means you are losing some level of control.
+You are also losing direct oversight. As a result performance feedback for InnerSource contributors needs to take the perspective of the host team into account.
+
+* Impact of language: InnerSource and Agile uses some of the same tooling - for different purposes.
```

Comment:
`Product Owner` in InnerSource needs to negotiate and align on off-team contributions (not just work for the host team).

Reply:
I've tried to address that in the negotiation part.

Reply:
OK maybe just at least a mention of it, then?  It sounds like a common misunderstanding and so seems appropriate to mention here even if it is explained in detail somewhere else.

```suggestion
* Impact of language: InnerSource and Agile uses some of the same tooling - for different purposes.
   - Product Owner needs to support and align on off-team contributions (rather than strictly work with the Host team).
More to come on this in **InnerSource and Negotiation**.
```